docs/arcadia-users-group/20240416-intro-to-ml1/make_peptides_ml_dataset.md

```diff
@@ -0,0 +1,278 @@
+   ```
+
+6. Filter to large clusters composed of peptides that all have the same bioactivity prediction and write a TSV file.
+   Many of the peptides that end up in clusters have multiple bioactivity predictions (e.g. antimicrobial and therapeutic).
+   The script below filters to peptide clusters that all have at least one bioactivity label that is the same.
+   It also adds additional metadata (GO, PFAM) from the SQL database to the data.frame of peptide information.
+   ```r
+   library(tidyverse)
+   library(janitor)
+   
+   # The goal of this script is to create a demo machine learning data set from peptides and their metadata.
+   # It filters peptides in the peptipedia database to large clusters of similar peptides
+   # (80% identity or greater) that all have the same bioactivity label.
+   # It then joins selected peptides to their metadata.
+   
+   # filter to peptides clusters with many members that have the same bioactivity --------
+
+   # read in cluster information from mmseqs2 linclust
+   clusters <- read_tsv("~/Downloads/backup_peptipedia_29_03_2023/mmseqs/peptipedia_minlen10-0.8_cluster.tsv",
+                        col_names = c("rep", "member"))
+   
+   # read in peptide bioactivity information
+   peptide_bioactivity <- read_csv('~/Downloads/backup_peptipedia_29_03_2023/peptipedia.csv') %>%
+     clean_names()
+   
+   # make the data set a little bit smaller by filtering to clusters that have > 5 members
+   # otherwise, the lines of code to find clusters with all of the same bioactivity takes too long
+   cluster_sizes <- clusters %>%
+     group_by(rep) %>%
+     tally() %>%
+     arrange(desc(n))
+   
+   large_clusters <- cluster_sizes %>%
+     filter(n > 2)
+   
+   clusters_filtered <- clusters %>%
+     filter(rep %in% large_clusters$rep)
+   
+   # filter the bioactivity data set to match
+   peptide_bioactivity_filtered <- peptide_bioactivity %>%
+     filter(idpeptide %in% clusters_filtered$member)
+   
+   # filter to clusters that all have the same bioactivity for at least one bioactivity label
+   # (e.g. all peptides have a predicted bioactivity of "antimicrobial" in a given cluster)
+   peptide_bioactivity_with_clusters <- left_join(clusters_filtered, peptide_bioactivity_filtered,
+                                                  by = c("member" = "idpeptide"))
+   
+   peptide_bioactivity_with_clusters_long <- peptide_bioactivity_with_clusters %>%
+     pivot_longer(cols = starts_with("allergen"):last_col(), names_to = "bioactivity", values_to = "value")
+   
+   # Find clusters where all members have a 1 for any bioactivity
+   clusters_all_1_any_bioactivity <- peptide_bioactivity_with_clusters_long %>%
+     group_by(rep, bioactivity) %>%
+     summarise(all_ones = all(value == 1), .groups = 'drop') %>%
+     filter(all_ones) %>%
+     distinct(rep, .keep_all = TRUE)
+   
+   # filter to those clusters where all members have at least 1 bioactivity the same
```

Comment:
nit: "have at least 1 bioactivity the same" reads odd

Reply:
fixed!